File: src/constants/theme.ts
// Bloom — Botanical Warm Theme
// Warm cream, terracotta, sage green, golden amber
// Inspired by a sunlit garden journal

export const colors = {
  // Brand
  primary: {
    50: '#FBF0EC',
    100: '#F2D6CC',
    200: '#E4AD99',
    300: '#D4937E',
    400: '#C4725A',
    500: '#A45B44',
    600: '#8B4A37',
    700: '#6E3A2B',
  },
  secondary: {
    50: '#F0F5EF',
    100: '#DAE7D8',
    200: '#C0D6BD',
    300: '#A8C4A5',
    400: '#8BA888',
    500: '#6B8B6A',
    600: '#557055',
    700: '#3F5440',
  },
  accent: {
    50: '#FEF7E8',
    100: '#FCEBC4',
    200: '#F5D48A',
    300: '#F0C478',
    400: '#E8A948',
    500: '#D49330',
    600: '#B57A20',
  },
  neutral: {
    50: '#FAF6F0',
    100: '#F3EDE6',
    200: '#E8E0D8',
    300: '#D6CCC2',
    400: '#B8AFA6',
    500: '#7A7570',
    600: '#5C5752',
    700: '#44403C',
    800: '#2D2926',
    900: '#1C1917',
  },
  error: '#C75450',
  warning: '#D49330',
  success: '#6B8B6A',
  info: '#7E9EB5',
  background: '#FAF6F0',
  surface: '#FFFFFF',
};

export const fonts = {
  serif: 'DMSerifDisplay_400Regular',
  sans: 'DMSans_400Regular',
  sansMedium: 'DMSans_500Medium',
  sansSemiBold: 'DMSans_600SemiBold',
  sansBold: 'DMSans_700Bold',
  rounded: 'Nunito_600SemiBold',
  roundedBold: 'Nunito_700Bold',
};

export const typography = {
  // Serif display — screen titles, emotional headers
  displayLarge: {
    fontFamily: fonts.serif,
    fontSize: 32,
    lineHeight: 40,
  } as const,
  displayMedium: {
    fontFamily: fonts.serif,
    fontSize: 26,
    lineHeight: 34,
  } as const,
  displaySmall: {
    fontFamily: fonts.serif,
    fontSize: 22,
    lineHeight: 30,
  } as const,
  // Sans — headers, body, labels
  heading1: {
    fontFamily: fonts.sansBold,
    fontSize: 24,
    lineHeight: 32,
  } as const,
  heading2: {
    fontFamily: fonts.sansSemiBold,
    fontSize: 20,
    lineHeight: 28,
  } as const,
  heading3: {
    fontFamily: fonts.sansSemiBold,
    fontSize: 17,
    lineHeight: 24,
  } as const,
  body: {
    fontFamily: fonts.sans,
    fontSize: 15,
    lineHeight: 22,
  } as const,
  bodyBold: {
    fontFamily: fonts.sansSemiBold,
    fontSize: 15,
    lineHeight: 22,
  } as const,
  caption: {
    fontFamily: fonts.sansMedium,
    fontSize: 13,
    lineHeight: 18,
  } as const,
  small: {
    fontFamily: fonts.sansMedium,
    fontSize: 11,
    lineHeight: 16,
  } as const,
  // Rounded — tags, badges, mood labels
  tag: {
    fontFamily: fonts.rounded,
    fontSize: 12,
    lineHeight: 16,
  } as const,
};

export const spacing = {
  xs: 4,
  sm: 8,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 48,
};

export const borderRadius = {
  sm: 8,
  md: 14,
  lg: 20,
  xl: 24,
  full: 9999,
};

export const shadows = {
  sm: {
    shadowColor: colors.neutral[800],
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.04,
    shadowRadius: 8,
    elevation: 2,
  },
  md: {
    shadowColor: colors.neutral[800],
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.06,
    shadowRadius: 12,
    elevation: 3,
  },
  lg: {
    shadowColor: colors.neutral[800],
    shadowOffset: { width: 0, height: 8 },
    shadowOpacity: 0.08,
    shadowRadius: 24,
    elevation: 5,
  },
  warm: {
    shadowColor: colors.primary[400],
    shadowOffset: { width: 0, height: 6 },
    shadowOpacity: 0.2,
    shadowRadius: 16,
    elevation: 6,
  },
};

File: ios-native/Bloom/DesignSystem/Theme.swift
import SwiftUI

/// Sistema de diseño de Bloom — paleta botánica cálida.
/// Portado de `src/constants/theme.ts` de la app React Native.
enum Theme {

    // MARK: - Colores

    enum Palette {
        // Brand — terracota
        static let primary50 = Color(hex: "FBF0EC")
        static let primary100 = Color(hex: "F2D6CC")
        static let primary200 = Color(hex: "E4AD99")
        static let primary300 = Color(hex: "D4937E")
        static let primary400 = Color(hex: "C4725A")
        static let primary500 = Color(hex: "A45B44")
        static let primary600 = Color(hex: "8B4A37")
        static let primary700 = Color(hex: "6E3A2B")

        // Verde salvia
        static let secondary50 = Color(hex: "F0F5EF")
        static let secondary100 = Color(hex: "DAE7D8")
        static let secondary200 = Color(hex: "C0D6BD")
        static let secondary300 = Color(hex: "A8C4A5")
        static let secondary400 = Color(hex: "8BA888")
        static let secondary500 = Color(hex: "6B8B6A")
        static let secondary600 = Color(hex: "557055")
        static let secondary700 = Color(hex: "3F5440")

        // Ámbar dorado
        static let accent50 = Color(hex: "FEF7E8")
        static let accent100 = Color(hex: "FCEBC4")
        static let accent200 = Color(hex: "F5D48A")
        static let accent300 = Color(hex: "F0C478")
        static let accent400 = Color(hex: "E8A948")
        static let accent500 = Color(hex: "D49330")
        static let accent600 = Color(hex: "B57A20")

        // Neutros cálidos
        static let neutral50 = Color(hex: "FAF6F0")
        static let neutral100 = Color(hex: "F3EDE6")
        static let neutral200 = Color(hex: "E8E0D8")
        static let neutral300 = Color(hex: "D6CCC2")
        static let neutral400 = Color(hex: "B8AFA6")
        static let neutral500 = Color(hex: "7A7570")
        static let neutral600 = Color(hex: "5C5752")
        static let neutral700 = Color(hex: "44403C")
        static let neutral800 = Color(hex: "2D2926")
        static let neutral900 = Color(hex: "1C1917")

        // Semánticos
        static let error = Color(hex: "C75450")
        static let warning = Color(hex: "D49330")
        static let success = Color(hex: "6B8B6A")
        static let info = Color(hex: "7E9EB5")
        static let background = Color(hex: "FAF6F0")
        static let surface = Color.white
    }

    // MARK: - Espaciado

    enum Spacing {
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 16
        static let lg: CGFloat = 24
        static let xl: CGFloat = 32
        static let xxl: CGFloat = 48
    }

    // MARK: - Radios

    enum Radius {
        static let sm: CGFloat = 8
        static let md: CGFloat = 14
        static let lg: CGFloat = 20
        static let xl: CGFloat = 24
        static let full: CGFloat = 9999
    }

    // MARK: - Sombras

    struct Shadow {
        let color: Color
        let radius: CGFloat
        let x: CGFloat
        let y: CGFloat

        static let sm = Shadow(color: Palette.neutral800.opacity(0.04), radius: 8, x: 0, y: 2)
        static let md = Shadow(color: Palette.neutral800.opacity(0.06), radius: 12, x: 0, y: 4)
        static let lg = Shadow(color: Palette.neutral800.opacity(0.08), radius: 24, x: 0, y: 8)
        static let warm = Shadow(color: Palette.primary400.opacity(0.2), radius: 16, x: 0, y: 6)
    }
}

extension View {
    /// Aplica una sombra del sistema de diseño.
    func bloomShadow(_ shadow: Theme.Shadow) -> some View {
        self.shadow(color: shadow.color, radius: shadow.radius, x: shadow.x, y: shadow.y)
    }
}
